sympy/polys/tests/test_polymatrix.py

```diff
@@ -0,0 +1,39 @@
+from sympy.matrices.dense import Matrix
+from sympy.polys.polymatrix import PolyMatrix
+from sympy.polys import Poly
+
+from sympy import S
+
+from sympy.abc import x
+
+
+def test_polymatrix():
+    pm1 = PolyMatrix([[Poly(x**2, x), Poly(-x, x)], [Poly(x**3, x), Poly(-1 + x, x)]])
+    v1 = PolyMatrix([[1, 0], [-1, 0]])
+    m1 = Matrix([[1, 0], [-1, 0]])
+    A = PolyMatrix([[Poly(x**2 + x, x), Poly(0, x)], \
+                        [Poly(x**3 - x + 1, x), Poly(0, x)]])
+    B = PolyMatrix([[Poly(x**2, x), Poly(-x, x)], [Poly(-x**2, x), Poly(x, x)]])
+    assert isinstance(pm1*v1, PolyMatrix)
+    assert pm1*v1 == A
+    assert pm1*m1 == A
+    assert v1*pm1 == B
+    assert m1*pm1 == B
+
+    pm2 = PolyMatrix([[Poly(x**2, x, domain='QQ'), Poly(0, x, domain='QQ'), Poly(-x**2, x, domain='QQ'), \
+                    Poly(x**3, x, domain='QQ'), Poly(0, x, domain='QQ'), Poly(-x**3, x, domain='QQ')]])
+    v2 = PolyMatrix([1, 0, 0, 0, 0, 0])
+    m2 = Matrix([1, 0, 0, 0, 0, 0])
+    C = PolyMatrix([[Poly(x**2, x, domain='QQ')]])
+    assert pm2*v2 == C
+    assert pm2*m2 == C
+
+
+def test_polymatrix_mul():
+    pm = PolyMatrix([[Poly(x**2, x), S(1)]])
+    assert (S(1)/2)*pm == PolyMatrix([[Poly(1/2*x**2, x, domain='QQ'), 1]])
+    assert pm*(S(1)/2) == PolyMatrix([[Poly(1/2*x**2, x, domain='QQ'), 1]])
+
+    pm2 = PolyMatrix([[Poly(x**2, x, domain='ZZ'), Poly(-x**2, x, domain='ZZ')]])
+    v2 = Matrix([1, -1])
+    assert pm2*v2 == PolyMatrix([[Poly(2*x**2, x, domain='ZZ')]])
```

Comment:
I don't think we make `v2*pm2` to evaluate such that it returns containing `Poly(x**2, x)` instead of `-Poly(-x**2, x)`.